zschema/leaves.py

```diff
@@ -246,16 +243,24 @@ class Enum(Leaf):
     INVALID = 23
     VALID = None
 
-    def __init__(self, values=[None,], *args, **kwargs):
+    def __init__(self, values=None, *args, **kwargs):
         Leaf.__init__(self, *args, **kwargs)
+        if values is None:
+            values = []
         self.values = values
         self.values_s = set(values)
 
     def _validate(self, name, value):
-        if value not in self.values_s:
+        if len(self.values_s) and value not in self.values_s:
             m = "%s: the value %s is not a valid enum option" % (name, value)
             raise DataValidationException(m)
 
+    def _docs_common(self, parent_category):
+        retv = super(Enum, self)._docs_common(parent_category)
+        if len(self.values_s):
+            retv["values"] = list(self.values_s)
```

Comment:
I don't think I understand why you convert to a list here given that L250 sets the list locally?

Reply:
This seems particularly worrisome because enums are inherently ordered.

Reply:
> I don't think I understand why you convert to a list here given that L250 sets the list locally.

The list is available locally, but it would be confusing to present to a documentation user a list of possible values which might contain duplicates. Using a set as canon for documentation of the enum's possible values avoids that confusion.

> This seems particularly worrisome because enums are inherently ordered.

Enumerated types in general are not inherently ordered; they're a set of identifiers. In some languages which allow (or require) an enum to be backed by (or convertible to) an integral type, the order in which the identifiers are listed *does* matter.

ZSchema is not a language where this ordering matters; we use the schema-provided list of identifiers to document and validate which values are allowed in an `Enum` leaf. These are both use cases which are well-served by a set; we don't actually use the order anywhere.

(The fact that the schema declares possible `Enum` values in a list is an artifact of API convenience.)